client/src/pages/Resource/index.js

```diff
@@ -0,0 +1,78 @@
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+
+import { ResourceService, useService } from "../../services";
+
+import "./Resource.scss";
+
+export default function Resource() {
+	const [resource, setResource] = useState(undefined);
+	const resourceService = useService(ResourceService);
+	const { id } = useParams();
+
+	useEffect(() => {
+		resourceService.getOne(id).then(setResource);
+	}, [id, resourceService]);
+
+	if (!resource) {
+		return null;
+	}
```

Comment:
❓ Wasn't it at this point that you had returned the **NotFoundError Page** for this particular missing resource?

Reply:
❓ I'm not sure quite what you're asking here. This is effectively the "loading" state, we don't want to immediately send the user to a different place because `!resource` is true _before_ the fetch succeeds. We don't expect users to navigate to a random resource page (they _could_, but most people _don't_!), they're coming via a link so we know the resource will most probably exist.

Reply:
💡  Specifically, I was commenting on the **return value** while `!resource` is true, since I expected a 404 error page to be displayed rather than a **blank page in production** or an **error in development**.

Reply:
The scope of the current story is to show details of _published_ resources. From an _agile development_ perspective, we should focus on delivering a thin slice reflected by the user story. This page isn't reachable (through normal means, i.e clicking links, rather than editing the URL directly) _except_ where we're sure there's a published resource; the link to drafts was an error, as you've already pointed out, and we don't show links to a non-existing resource anywhere.

Would it be better UX to show an explicit loading state, then an error message if they've somehow got to this page with a non-existent or draft resource? Yes, absolutely.

But is that more valuable than whatever's next in the backlog? That's less clear, and not really the devs' responsibility to decide. In a mature product you might have a "definition of done" that effectively adds more acceptance criteria (must show loading state, must show error messages, etc.) but in the absence of that I would (if we didn't already have it) add a _new story_ to the PM's inbox and let them know - that gives them the option of either saying "yes that's important do it next" or "actually I'd rather you did this other story instead" (through the mechanism of prioritising and reordering the backlog).

(Responding also to [this comment](https://github.com/Emmanuel-Dominic/tech-products-demo/pull/2#discussion_r1481339848) as I think they're fairly coupled.)